liblog/tests/log.cpp

```diff
@@ -0,0 +1,205 @@
+#include "log.hpp"
+#include "doctest.h"
+#include "log_test.hpp"
+#include "utils.hpp"
+#include <boost/regex.hpp>
+#include <fstream>
+#include <future>
+#include <iostream>
+#include <sstream>
+using namespace std::string_literals;
+
+TEST_SUITE_BEGIN("liblog");
+
+TEST_CASE("liblog - log should write to a ostream correctly.") {
+	auto file = std::make_shared<std::ostringstream>();
+	logger::set_output_location(file);
+
+	LOG(note, "{}", "hello");
+	boost::smatch m;
+	std::string logged = file->str();
+	CHECK(boost::regex_match(logged, m, log_regex));
+	REQUIRE_EQ(m.size(), 5);
+	CHECK_EQ(m[1], "note"s);
+	// LIBLOG_DEBUG is not defined time and file location should be empty strings.
+	CHECK_EQ(m[2], ""s);
+	CHECK_EQ(m[3], ""s);
+	CHECK_EQ(m[4], "hello\n"s);
+}
+
+TEST_CASE("liblog - should be able to change current severity") {
+	auto file = std::make_shared<std::ostringstream>();
+	logger::set_output_location(file);
+	// Default severity is note.
+	CHECK_EQ(logger::current_severity.get(), logger::severity::note);
+	logger::current_severity.set(logger::severity::fatal_error);
+	CHECK_EQ(logger::current_severity.get(), logger::severity::fatal_error);
+
+	// Reset current severity to note.
+	logger::current_severity.set(logger::severity::note);
+}
+
+TEST_CASE("liblog - log should not print if given severitiy is lower than current severity") {
+	auto file = std::make_shared<std::ostringstream>();
+	logger::set_output_location(file);
+
+	logger::severity s = logger::severity::note;
+
+	SUBCASE("debug") {
+		s = logger::severity::debug;
+	}
+	SUBCASE("info") {
+		s = logger::severity::info;
+	}
+	SUBCASE("note") {
+		s = logger::severity::note;
+	}
+	SUBCASE("warning") {
+		s = logger::severity::warning;
+	}
+	SUBCASE("error") {
+		s = logger::severity::error;
+	}
+	SUBCASE("fatal_error") {
+		s = logger::severity::fatal_error;
+	}
+
+	logger::current_severity.set(s);
+
+	LOG(warning, "{}", "hello");
+	std::string logged = file->str();
+	boost::smatch m;
+	if (s == logger::severity::debug || s == logger::severity::info || s == logger::severity::note
+	    || s == logger::severity::warning) {
+		CHECK(boost::regex_match(logged, m, log_regex));
+		REQUIRE_EQ(m.size(), 5);
+	}
+	else {
+		CHECK_EQ(logged, ""s);
+	}
+	// Reset current severity to note.
+	logger::current_severity.set(logger::severity::note);
+}
+
+TEST_CASE("liblog - should be able to write to a file.") {
+	std::string file_name = create_temp_file();
+	logger::set_output_location(file_name);
+	LOG(fatal_error, "{}", "hello");
+	// Close file by opening a different stream.
+	auto temp_loc = std::make_shared<std::ostringstream>();
+	logger::set_output_location(temp_loc);
+
+	std::fstream file(file_name);
```

Comment:
This file should be deleted. We have `boost::filesystem` available, so you can use that.